mlir/include/mlir/Dialect/SCF/IR/SCF.h

```diff
@@ -107,6 +107,17 @@ LoopNest buildLoopNest(OpBuilder &builder, Location loc, ValueRange lbs,
                        function_ref<void(OpBuilder &, Location, ValueRange)>
                            bodyBuilder = nullptr);
 
+/// Perform a replacement of one iter OpOperand of an scf.for to the
+/// `replacement` value with a different type. A callback is used to insert
+/// cast ops inside the block to account for type differences.
+using ValueTypeCastFnTy =
+    std::function<Value(OpBuilder &, Location loc, Type, Value)>;
+SmallVector<Value> replaceAndCastForOpIterArg(RewriterBase &rewriter,
+                                              scf::ForOp forOp,
+                                              OpOperand &operand,
```

Comment:
Do we need to pass it by reference? I'd avoid it if we are not modifying the object. It looks like it can just be `OpOperand` or `const OpOperand &` type. I think `OpOperand` is enough because it is a pointer-like type.

Reply:
There's no copy constructor for `OpOperand` and `operand.getOperandNumber()` is non-const so I think it has to be `OpOperand &`. Maybe it is a bug that `getOperandNumber` is non-const, but I'm not familiar with that code.